api/ai/algorithm_runner.py

```diff
@@ -52,12 +54,25 @@ def __init__(
     def generate(self, students: List[Student]) -> TeamSet:
         # the algorithm classes internally track generated teams, so a new instance of the
         #   algorithm class MUST be created to run a new generation without side effects
+        # This generated instance is initialized in the prepare method
+        return self.algorithm.generate(students)
+
+    def prepare(self, students: List[Student]) -> None:
```

Comment:
Since I don't think `prepare` should be excluded from runtime, I think there's no need for `AlgorithmRunner.prepare`. 

Instead, I propose that `AlgorithmRunner.generate` just calls `algorithm.prepare` and then `algorithm.generate`